virttest/utils_misc.py

```diff
@@ -2756,14 +2756,25 @@ def format_linux_disk(session, did, mountpoint, size, fstype="ext3"):
     for disk in disks.splitlines():
         if did in disk:
             kname = disk.split()[0]
+            list_parttable_cmd = "parted -s %s print" % kname
+            mklabel_cmd = "parted -s %s mklabel msdos" % kname
+            output = session.cmd_output(list_parttable_cmd)
+            if "Error" in output:
+                logging.info("Create partition table on disk '%s'" % kname)
+                session.cmd(mklabel_cmd)
             logging.info("Create partition on disk '%s'" % kname)
             mkpart_cmd = "parted -s %s  mkpart primary 0 %s" % (kname, size)
             session.cmd(mkpart_cmd)
+            time.sleep(5)
```

Comment:
1. I think seesion.cmd() will got return when "mkport_cmd" complete, so sleep not necessary.
2. Better to execute "partprobe"  after change partition label.